~I'm not sure why this failed the second time.~
My Xcode is letting anything compile when it shouldn't. Need a restart.
The `TransitionAnimationType` enums are missing their labels.

```** BUILD FAILED **
The following build commands failed:
	CompileSwift normal x86_64 /Users/travis/build/IBAnimatable/IBAnimatable/IBAnimatableApp/TransitionTableViewController.swift
	CompileSwiftSources normal x86_64 com.apple.xcode.tools.swift.compiler
(2 failures)

The command "xcodebuild build -project IBAnimatable.xcodeproj -scheme IBAnimatableApp -sdk iphonesimulator -destination 'platform=iOS Simulator,name=iPhone 6,OS=latest'" exited with 65.
3.08s$ bundle exec danger

⚠️ 1 Warning. Don't worry, everything is fixable.
Danger does not have write access to the PR to set a PR status.
Results:
Warnings:
- [ ] Consider adding supporting documentation to this change. Documentation can be found in the `docs` directory.

The command "bundle exec danger" exited with 0.
Done. Your build exited with 1.
/Users/travis/.travis/job_stages: line 150: shell_session_update: command not found
```

--- a/IBAnimatable/TransitionAnimationType.swift
+++ b/IBAnimatable/TransitionAnimationType.swift
@@ -305,8 +305,7 @@ extension TransitionAnimationType {
       switch self {
       case .close: return .open
       case .open: return .close
-      case .none:
-        fatalError("No opposing HollowState for HollowState.none")
+      case .none: return .none
       }
     }
 

--- a/IBAnimatableApp/TransitionTableViewController.swift
+++ b/IBAnimatableApp/TransitionTableViewController.swift
@@ -48,92 +48,133 @@ private extension TransitionTableViewController {
 
     transitionAnimationsHeaders.append("Fade")
 
-    let fadeAnimations: [TransitionAnimationType] = [.fade(direction: .in), .fade(direction: .out), .fade(direction: .cross)]
+    let fadeAnimations: [TransitionAnimationType] = [.fade(direction: .in),
+                                                     .fade(direction: .out),
+                                                     .fade(direction: .cross)]
 
     transitionAnimations.append(toString(animations: fadeAnimations))
 
     transitionAnimationsHeaders.append("SystemCube")
-    let cubeAnimations: [TransitionAnimationType] = [
-      .systemCube(from: .left), .systemCube(from: .right), .systemCube(from: .top), .systemCube(from: .bottom)]
+
+    let cubeAnimations: [TransitionAnimationType] = [.systemCube(from: .left),
+                                                     .systemCube(from: .right),
+                                                     .systemCube(from: .top),
+                                                     .systemCube(from: .bottom)]
 
     transitionAnimations.append(toString(animations: cubeAnimations))
 
     transitionAnimationsHeaders.append("SystemFlip")
-    let flipSystemAnimations: [TransitionAnimationType] = [
-      .systemFlip(from: .left), .systemFlip(from: .right), .systemFlip(from: .top), .systemFlip(from: .bottom)]
+
+    let flipSystemAnimations: [TransitionAnimationType] = [.systemFlip(from: .left),
+                                                           .systemFlip(from: .right),
+                                                           .systemFlip(from: .top),
+                                                           .systemFlip(from: .bottom)]
 
     transitionAnimations.append(toString(animations: flipSystemAnimations))
 
     transitionAnimationsHeaders.append("SystemMoveIn")
-    let moveAnimations: [TransitionAnimationType] = [
-      .systemMoveIn(from: .left), .systemMoveIn(from: .right), .systemMoveIn(from: .top), .systemMoveIn(from: .bottom)]
+
+    let moveAnimations: [TransitionAnimationType] = [.systemMoveIn(from: .left),
+                                                     .systemMoveIn(from: .right),
+                                                     .systemMoveIn(from: .top),
+                                                     .systemMoveIn(from: .bottom)]
 
     transitionAnimations.append(toString(animations: moveAnimations))
 
     transitionAnimationsHeaders.append("SystemPush")
-    let pushAnimations: [TransitionAnimationType] = [
-      .systemPush(from: .left), .systemPush(from: .right), .systemMoveIn(from: .top), .systemMoveIn(from: .bottom)]
+
+    let pushAnimations: [TransitionAnimationType] = [.systemPush(from: .left),
+                                                     .systemPush(from: .right),
+                                                     .systemMoveIn(from: .top),
+                                                     .systemMoveIn(from: .bottom)]
 
     transitionAnimations.append(toString(animations: pushAnimations))
 
     transitionAnimationsHeaders.append("SystemReveal")
-    let revealAnimations: [TransitionAnimationType] = [
-      .systemReveal(from: .left), .systemReveal(from: .right), .systemReveal(from: .top), .systemReveal(from: .bottom)]
+
+    let revealAnimations: [TransitionAnimationType] = [.systemReveal(from: .left),
+                                                       .systemReveal(from: .right),
+                                                       .systemReveal(from: .top),
+                                                       .systemReveal(from: .bottom)]
 
     transitionAnimations.append(toString(animations: revealAnimations))
 
     transitionAnimationsHeaders.append("SystemPage")
+
     let pageAnimations: [TransitionAnimationType] = [.systemPage(type: .curl), .systemPage(type: .unCurl)]
 
     transitionAnimations.append(toString(animations: pageAnimations))
 
     transitionAnimationsHeaders.append("SystemCameraIris")
-    let cameraAnimations: [TransitionAnimationType] = [
-      .systemCameraIris(hollowState: .none), .systemCameraIris(hollowState: .open), .systemCameraIris(hollowState: .close)]
+
+    let cameraAnimations: [TransitionAnimationType] = [.systemCameraIris(hollowState: .none),
+                                                       .systemCameraIris(hollowState: .open),
+                                                       .systemCameraIris(hollowState: .close)]
 
     transitionAnimations.append(toString(animations: cameraAnimations))
 
     transitionAnimationsHeaders.append("Fold")
-    let foldAnimations: [TransitionAnimationType] = [
-      .fold(from: .left, folds: nil), .fold(from: .right, folds: nil), .fold(from: .top, folds: nil), .fold(from: .bottom, folds: nil)]
+
+    let foldAnimations: [TransitionAnimationType] = [.fold(from: .left, folds: nil),
+                                                     .fold(from: .right, folds: nil),
+                                                     .fold(from: .top, folds: nil),
+                                                     .fold(from: .bottom, folds: nil)]
 
     transitionAnimations.append(toString(animations: foldAnimations))
 
     transitionAnimationsHeaders.append("Portal")
-    let portalAnimations: [TransitionAnimationType] = [.portal(direction: .forward, zoomScale: 0.3), .portal(direction: .backward, zoomScale: nil)]
+
+    let portalAnimations: [TransitionAnimationType] = [.portal(direction: .forward, zoomScale: 0.3),
+                                                       .portal(direction: .backward, zoomScale: nil)]
 
     transitionAnimations.append(toString(animations: portalAnimations))
 
     transitionAnimationsHeaders.append("NatGeo")
+
     let natGeoAnimations: [TransitionAnimationType] = [.natGeo(to: .left), .natGeo(to: .right)]
+
     transitionAnimations.append(toString(animations: natGeoAnimations))
 
     transitionAnimationsHeaders.append("Turn")
-    let turnAnimations: [TransitionAnimationType] = [.turn(from: .left), .turn(from: .right), .turn(from: .top), .turn(from: .bottom)]
+
+    let turnAnimations: [TransitionAnimationType] = [.turn(from: .left),
+                                                     .turn(from: .right),
+                                                     .turn(from: .top),
+                                                     .turn(from: .bottom)]
+
     transitionAnimations.append(toString(animations: turnAnimations))
 
     transitionAnimationsHeaders.append("Cards")
+
     let cardAnimations: [TransitionAnimationType] = [.cards(direction: .forward), .cards(direction: .backward)]
+
     transitionAnimations.append(toString(animations: cardAnimations))
 
     transitionAnimationsHeaders.append("Flip")
+
     let flipAnimations: [TransitionAnimationType] = [.flip(from: .left), .flip(from: .right)]
+
     transitionAnimations.append(toString(animations: flipAnimations))
 
     transitionAnimationsHeaders.append("Slide")
-    let slideAnimations: [TransitionAnimationType] = [
-      .slide(to: .left, isFade: true), .slide(to: .right, isFade: false), .slide(to: .top, isFade: true), .slide(to: .bottom, isFade: false)]
+
+    let slideAnimations: [TransitionAnimationType] = [.slide(to: .left, isFade: true),
+                                                      .slide(to: .right, isFade: false),
+                                                      .slide(to: .top, isFade: true),
+                                                      .slide(to: .bottom, isFade: false)]
 
     transitionAnimations.append(toString(animations: slideAnimations))
 
     transitionAnimationsHeaders.append("Others")
-    let otherAnimations: [TransitionAnimationType] = [
-      .systemRotate, .systemRippleEffect, .explode(xFactor: nil, minAngle: nil, maxAngle: nil), .explode(xFactor: 10, minAngle: -20, maxAngle: 20)]
+    let otherAnimations: [TransitionAnimationType] = [.systemRotate,
+                                                      .systemRippleEffect,
+                                                      .explode(xFactor: nil, minAngle: nil, maxAngle: nil),
+                                                      .explode(xFactor: 10, minAngle: -20, maxAngle: 20)]
 
     transitionAnimations.append(toString(animations: otherAnimations))
   }
 
-  func toString(animations: [TransitionAnimationType]) -> [String] {
+  private func toString(animations: [TransitionAnimationType]) -> [String] {
     return animations.map { $0.asString }
   }
 

--- a/IBAnimatableApp/TransitionTableViewController.swift
+++ b/IBAnimatableApp/TransitionTableViewController.swift
@@ -185,82 +185,47 @@ fileprivate extension TransitionAnimationType {
   var asString: String {
     switch self {
     case .fade(let direction):
-      return "Fade" + "(\(direction.asString))"
+      return "fade" + "(\(direction.rawValue)"
     case .systemCube(let direction):
-      return "SystemCube" + "(\(direction.asString)"
+      return "systemCube" + "(\(direction.rawValue)"
     case .systemFlip(let direction):
-      return "SystemFlip" + "(\(direction.asString))"
+      return "systemFlip" + "(\(direction.rawValue))"
     case .systemMoveIn(let direction):
-      return "SystemMoveIn" + "(\(direction.asString))"
+      return "systemMoveIn" + "(\(direction.rawValue))"
     case .systemPush(let direction):
-      return "SystemPush" + "(\(direction.asString))"
+      return "systemPush" + "(\(direction.rawValue))"
     case .systemReveal(let direction):
-      return "SystemReveal" + "(\(direction.asString))"
+      return "systemReveal" + "(\(direction.rawValue))"
     case .systemPage(let type):
-      return "SystemPage(\(type.asString))"
+      return "systemPage(\(type.rawValue))"
     case .systemCameraIris(let hollowState):
-      return hollowState == .none ? "SystemCameraIris" : "SystemCameraIris" + "(\(hollowState.asString))"
+      return "systemCameraIris" + (hollowState == .none ? "" : "(\(hollowState.rawValue))")
     case .fold(let direction, _):
-      return "Fold" + "(\(direction.asString))"
+      return "fold" + "(\(direction.rawValue))"
     case .portal(let direction, let zoomScale):
-      return zoomScale == nil ? "Portal" + "(\(direction.asString))" : "Portal" + "(\(direction.asString),\(zoomScale!))"
+      return "portal" + (zoomScale == nil ? "(\(direction.rawValue))" : "(\(direction.rawValue),\(zoomScale!))")
     case .natGeo(let direction):
-      return "NatGeo" + "(\(direction.asString))"
+      return "natGeo" + "(\(direction.rawValue))"
     case .turn(let direction):
-      return "Turn" + "(\(direction.asString))"
+      return "turn" + "(\(direction.rawValue))"
     case .cards(let direction):
-      return "Cards" + "(\(direction.asString))"
+      return "cards" + "(\(direction.rawValue))"
     case .flip(let direction):
-      return "Flip" + "(\(direction.asString))"
+      return "flip" + "(\(direction.rawValue))"
     case .slide(let direction, let isFade):
-      return isFade ? "Slide" + "(\(direction.asString), fade)" : "Slide" + "(\(direction.asString))"
+      return "slide" + (isFade ? "(\(direction.rawValue), fade)" : "slide" + "(\(direction.rawValue))")
     case .systemRotate:
-      return "SystemRotate"
+      return "systemRotate"
     case .systemRippleEffect:
-      return "SystemRippleEffect"
+      return "systemRippleEffect"
     case .systemSuckEffect:
-      return "SystemSuckEffect"
+      return "systemSuckEffect"
     case .explode(.some(let x), .some(let min), .some(let max)):
-      return "Explode" + "(\(x),\(min),\(max))"
+      return "explode" + "(\(x),\(min),\(max))"
     case .explode:
-      return "Explode"
+      return "explode"
     case .none:
-      return "None"
-    }
-  }
-}
-
-fileprivate extension TransitionAnimationType.Direction {
-  var asString: String {
-    switch self {
-    case .in: return "In"
-    case .out: return "Out"
-    case .left: return "Left"
-    case .right: return "Right"
-    case .top: return "Top"
-    case .bottom: return "Bottom"
-    case .forward: return "Forward"
-    case .backward: return "Backward"
-    case .cross: return "Cross"
-    }
-  }
-}
-
-fileprivate extension TransitionAnimationType.HollowState {
-  var asString: String {
-    switch self {
-    case .open: return "Open"
-    case .close: return "Close"
-    case .none: return ""
-    }
-  }
-}
-
-fileprivate extension TransitionAnimationType.PageType {
-  var asString: String {
-    switch self {
-    case .curl: return "Curl"
-    case .unCurl: return "UnCurl"
+      return "none"
     }
   }
 }

--- a/IBAnimatableApp/TransitionTableViewController.swift
+++ b/IBAnimatableApp/TransitionTableViewController.swift
@@ -45,44 +45,183 @@ class TransitionTableViewController: UITableViewController {
 private extension TransitionTableViewController {
 
   func populateTransitionTypeData() {
+
     transitionAnimationsHeaders.append("Fade")
-    transitionAnimations.append(["Fade", "Fade(In)", "Fade(Out)"])
+
+    let fadeAnimations: [TransitionAnimationType] = [.fade(direction: .in), .fade(direction: .out), .fade(direction: .cross)]
+
+    transitionAnimations.append(toString(animations: fadeAnimations))
+
     transitionAnimationsHeaders.append("SystemCube")
-    transitionAnimations.append(transitionTypeWithDirections(forName: "SystemCube"))
+    let cubeAnimations: [TransitionAnimationType] = [
+      .systemCube(from: .left), .systemCube(from: .right), .systemCube(from: .top), .systemCube(from: .bottom)]
+
+    transitionAnimations.append(toString(animations: cubeAnimations))
+
     transitionAnimationsHeaders.append("SystemFlip")
-    transitionAnimations.append(transitionTypeWithDirections(forName: "SystemFlip"))
+    let flipSystemAnimations: [TransitionAnimationType] = [
+      .systemFlip(from: .left), .systemFlip(from: .right), .systemFlip(from: .top), .systemFlip(from: .bottom)]
+
+    transitionAnimations.append(toString(animations: flipSystemAnimations))
+
     transitionAnimationsHeaders.append("SystemMoveIn")
-    transitionAnimations.append(transitionTypeWithDirections(forName: "SystemMoveIn"))
+    let moveAnimations: [TransitionAnimationType] = [
+      .systemMoveIn(from: .left), .systemMoveIn(from: .right), .systemMoveIn(from: .top), .systemMoveIn(from: .bottom)]
+
+    transitionAnimations.append(toString(animations: moveAnimations))
+
     transitionAnimationsHeaders.append("SystemPush")
-    transitionAnimations.append(transitionTypeWithDirections(forName: "SystemPush"))
+    let pushAnimations: [TransitionAnimationType] = [
+      .systemPush(from: .left), .systemPush(from: .right), .systemMoveIn(from: .top), .systemMoveIn(from: .bottom)]
+
+    transitionAnimations.append(toString(animations: pushAnimations))
+
     transitionAnimationsHeaders.append("SystemReveal")
-    transitionAnimations.append(transitionTypeWithDirections(forName: "SystemReveal"))
+    let revealAnimations: [TransitionAnimationType] = [
+      .systemReveal(from: .left), .systemReveal(from: .right), .systemReveal(from: .top), .systemReveal(from: .bottom)]
+
+    transitionAnimations.append(toString(animations: revealAnimations))
+
     transitionAnimationsHeaders.append("SystemPage")
-    transitionAnimations.append(["SystemPage(Curl)", "SystemPage(UnCurl)"])
+    let pageAnimations: [TransitionAnimationType] = [.systemPage(type: .curl), .systemPage(type: .unCurl)]
+
+    transitionAnimations.append(toString(animations: pageAnimations))
+
     transitionAnimationsHeaders.append("SystemCameraIris")
-    transitionAnimations.append(["SystemCameraIris", "SystemCameraIris(HollowOpen)", "SystemCameraIris(HollowClose)"])
+    let cameraAnimations: [TransitionAnimationType] = [
+      .systemCameraIris(hollowState: .none), .systemCameraIris(hollowState: .open), .systemCameraIris(hollowState: .close)]
+
+    transitionAnimations.append(toString(animations: cameraAnimations))
+
     transitionAnimationsHeaders.append("Fold")
-    transitionAnimations.append(transitionTypeWithDirections(forName: "Fold"))
+    let foldAnimations: [TransitionAnimationType] = [
+      .fold(from: .left, folds: nil), .fold(from: .right, folds: nil), .fold(from: .top, folds: nil), .fold(from: .bottom, folds: nil)]
+
+    transitionAnimations.append(toString(animations: foldAnimations))
+
     transitionAnimationsHeaders.append("Portal")
-    transitionAnimations.append(["Portal(Forward,0.3)", "Portal(Backward)"])
+    let portalAnimations: [TransitionAnimationType] = [.portal(direction: .forward, zoomScale: 0.3), .portal(direction: .backward, zoomScale: nil)]
+
+    transitionAnimations.append(toString(animations: portalAnimations))
+
     transitionAnimationsHeaders.append("NatGeo")
-    transitionAnimations.append(["NatGeo(Left)", "NatGeo(Right)"])
+    let natGeoAnimations: [TransitionAnimationType] = [.natGeo(to: .left), .natGeo(to: .right)]
+    transitionAnimations.append(toString(animations: natGeoAnimations))
+
     transitionAnimationsHeaders.append("Turn")
-    transitionAnimations.append(transitionTypeWithDirections(forName: "Turn"))
+    let turnAnimations: [TransitionAnimationType] = [.turn(from: .left), .turn(from: .right), .turn(from: .top), .turn(from: .bottom)]
+    transitionAnimations.append(toString(animations: turnAnimations))
+
     transitionAnimationsHeaders.append("Cards")
-    transitionAnimations.append(["Cards(Forward)", "Cards(Backward)"])
+    let cardAnimations: [TransitionAnimationType] = [.cards(direction: .forward), .cards(direction: .backward)]
+    transitionAnimations.append(toString(animations: cardAnimations))
+
     transitionAnimationsHeaders.append("Flip")
-    transitionAnimations.append(["Flip(Left)", "Flip(Right)"])
+    let flipAnimations: [TransitionAnimationType] = [.flip(from: .left), .flip(from: .right)]
+    transitionAnimations.append(toString(animations: flipAnimations))
+
     transitionAnimationsHeaders.append("Slide")
-    transitionAnimations.append(["Slide(Left, fade)", "Slide(Right)", "Slide(Top, fade)", "Slide(Bottom)"])
+    let slideAnimations: [TransitionAnimationType] = [
+      .slide(to: .left, isFade: true), .slide(to: .right, isFade: false), .slide(to: .top, isFade: true), .slide(to: .bottom, isFade: false)]
+
+    transitionAnimations.append(toString(animations: slideAnimations))
+
     transitionAnimationsHeaders.append("Others")
-    transitionAnimations.append(["SystemRotate", "SystemRippleEffect", "SystemSuckEffect", "Explode", "Explode(10,-20,20)"])
+    let otherAnimations: [TransitionAnimationType] = [
+      .systemRotate, .systemRippleEffect, .explode(xFactor: nil, minAngle: nil, maxAngle: nil), .explode(xFactor: 10, minAngle: -20, maxAngle: 20)]
+
+    transitionAnimations.append(toString(animations: otherAnimations))
   }
 
-  func transitionTypeWithDirections(forName prefixName: String) -> [String] {
-    return [prefixName + "(Left)", prefixName + "(Right)", prefixName + "(Top)", prefixName + "(Bottom)"]
+  func toString(animations: [TransitionAnimationType]) -> [String] {
+    return animations.map { $0.asString }
   }
 
+}
+
+fileprivate extension TransitionAnimationType {
+
+  var asString: String {
+    switch self {
+    case .fade(let direction):
+      return "Fade" + "(\(direction.asString))"
+    case .systemCube(let direction):
+      return "SystemCube" + "(\(direction.asString)"
+    case .systemFlip(let direction):
+      return "SystemFlip" + "(\(direction.asString))"
+    case .systemMoveIn(let direction):
+      return "SystemMoveIn" + "(\(direction.asString))"
+    case .systemPush(let direction):
+      return "SystemPush" + "(\(direction.asString))"
+    case .systemReveal(let direction):
+      return "SystemReveal" + "(\(direction.asString))"
+    case .systemPage(let type):
+      return "SystemPage(\(type.asString))"
+    case .systemCameraIris(let hollowState):
+      return hollowState == .none ? "SystemCameraIris" : "SystemCameraIris" + "(\(hollowState.asString))"
+    case .fold(let direction, _):
+      return "Fold" + "(\(direction.asString))"
+    case .portal(let direction, let zoomScale):
+      return zoomScale == nil ? "Portal" + "(\(direction.asString))" : "Portal" + "(\(direction.asString),\(zoomScale!))"
+    case .natGeo(let direction):
+      return "NatGeo" + "(\(direction.asString))"
+    case .turn(let direction):
+      return "Turn" + "(\(direction.asString))"
+    case .cards(let direction):
+      return "Cards" + "(\(direction.asString))"
+    case .flip(let direction):
+      return "Flip" + "(\(direction.asString))"
+    case .slide(let direction, let isFade):
+      return isFade ? "Slide" + "(\(direction.asString), fade)" : "Slide" + "(\(direction.asString))"
+    case .systemRotate:
+      return "SystemRotate"
+    case .systemRippleEffect:
+      return "SystemRippleEffect"
+    case .systemSuckEffect:
+      return "SystemSuckEffect"
+    case .explode(.some(let x), .some(let min), .some(let max)):
+      return "Explode" + "(\(x),\(min),\(max))"
+    case .explode:
+      return "Explode"
+    case .none:
+      return "None"
+    }
+  }
+}
+
+fileprivate extension TransitionAnimationType.Direction {
+  var asString: String {
+    switch self {
+    case .in: return "In"
+    case .out: return "Out"
+    case .left: return "Left"
+    case .right: return "Right"
+    case .top: return "Top"
+    case .bottom: return "Bottom"
+    case .forward: return "Forward"
+    case .backward: return "Backward"
+    case .cross: return "Cross"
+    }
+  }
+}
+
+fileprivate extension TransitionAnimationType.HollowState {
+  var asString: String {
+    switch self {
+    case .open: return "Open"
+    case .close: return "Close"
+    case .none: return ""
+    }
+  }
+}
+
+fileprivate extension TransitionAnimationType.PageType {
+  var asString: String {
+    switch self {
+    case .curl: return "Curl"
+    case .unCurl: return "UnCurl"
+    }
+  }
 }
 
 // MARK: - UITableViewDataSource / UITableViewDelegate

